MDANSE_GUI/Src/MDANSE_GUI/Tabs/Models/PlotDataModel.py

```diff
@@ -185,6 +204,42 @@ def add_file(self, filename: str):
             self._next_number += 1
             self.appendRow(new_item)
             new_item.populate(new_datafile._file)
+            self.finished_loading.emit(self._next_number - 1)
+            self.recent_files_update.emit(filename)
+
+    @Slot(str)
+    def recent_plot_files(self, plot_filepath: str):
+        """Adding recently loaded plot selection files to a json file. The json file is used to
+        populate the "Open Recent Plot Selection File" menu in the File menu.
+
+        Parameters
+        ----------
+        plot_filepath : Path of the added file
+        Returns
+        -------
+        Str
+            Recently loaded plot selection file path.
+        """
+        filename = self.DEFAULT_JSON_PATH
+        max_num_files = self.MAX_NUMBER_RECENT_FILES
+        if os.path.exists(filename) and os.path.getsize(filename) > 0:
+            with open(filename) as f:
+                recent_files = json.load(f)
+        else:
+            recent_files = []
```

Comment:
```suggestion
        filepath = Path(filename)
        if filepath.is_file():
            with filepath.open(encoding="utf-8") as file:
                recent_files = json.load(file)
        else:
            recent_files = []
```
Prefer using `Path`.